kernel/include/syscall_handler.h

```diff
@@ -125,6 +125,19 @@ extern void _thread_perms_all_clear(struct k_thread *thread);
  */
 void _k_object_uninit(void *obj);
 
+/**
+ * Initialize and reset permissions to only access by the caller
+ *
+ * Intended for scenarios where objects are fetched from slab pools
+ * and may have had different permissions set during prior usage.
+ *
+ * The object will be marked as initialized and the calling thread
+ * granted access to it.
+ *
+ * @param object Address of the kernel object
+ */
```

Comment:
This one is a little confusing.   I'm gathering that the idea is to take an object that you already have and "reinitialize" it so you don't have to go through the pair of k_obj_init()/uninit() calls?  If so, maybe document it like that?

But then I'm looking at k_object_init() and noticing that it doesn't reset the perms like this function does, and wondering whether that should have happened in the first place?

Basically the need for this makes me worried that maybe we have a messy API for object initialization state and should probably clean it up instead of adding new pseudoinitialization callls. :)

Reply:
Starting to write a lengthy reply, but probably better to just amend our documentation to explain the distinction between k_object_init() and k_object_recycle() as they are used for different purposes, so I'll do that instead.

Reply:
The doc has been expanded a bit.

Basically, for statically defined objects, you don't want this. If I declare a thread object toplevel, the permissions on it should persist even if the thread object is re-used.

However, for objects that are drawn from a pool, it needs it's permissions freshened up each time a pool member is returned to the user. It's past permission state needs to be wiped clean.